src/DocBlock/TypeExpression.php

```diff
@@ -208,8 +208,9 @@ final class TypeExpression
 
     private string $value;
 
-    private bool $isUnionType;
+    private bool $isGluedType;
 
+    /** @var '&'|'|'|'~' */
     private string $typesGlue;
```

Comment:
```suggestion
    /** @var null|'&'|'|'|'~' */
    private ?string $typesGlue = null;
```

This requires [changes in `PhpCsFixer\DocBlock\Annotation`](https://github.com/Wirone/PHP-CS-Fixer/commit/cc4ce62a75760d319743dbdcbb67582cc1a72898#diff-f9649c74d6fe25ca4b9adc1efa4582a68dad37b3c13176d8e3dd2b593db73e18) but makes the internals and public API more explicit - if the type is not composite, there's no glue.

Reply:
reverted completely because of `~` not implemented in this PR - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8210#discussion_r1765400782